pkg/domain/infra/abi/containers.go

```diff
@@ -1110,6 +1111,44 @@ func (ic *ContainerEngine) ContainerRun(ctx context.Context, opts entities.Conta
 		fmt.Fprintf(os.Stderr, "%s\n", w)
 	}
 
+	if opts.Spec != nil && !reflect.ValueOf(opts.Spec).IsNil() {
+		// If this is a checkpoint image, restore it.
+		img, resolvedImageName := opts.Spec.GetImage()
+		if img != nil && resolvedImageName != "" {
+			imgData, err := img.Inspect(ctx, nil)
+			if err != nil {
+				return nil, err
+			}
+			if imgData != nil {
+				_, isCheckpointImage := imgData.Annotations[define.CheckpointAnnotationRuntimeName]
+				if isCheckpointImage {
+					var restoreOptions entities.RestoreOptions
+					restoreOptions.Name = opts.Spec.Name
+					restoreOptions.Pod = opts.Spec.Pod
+					responses, err := ic.ContainerRestore(ctx, []string{resolvedImageName}, restoreOptions)
+					if err != nil {
+						return nil, err
+					}
+
+					report := entities.ContainerRunReport{}
+					for _, r := range responses {
+						report.Id = r.Id
+						report.ExitCode = 0
+						if r.Err != nil {
+							logrus.Errorf("Failed to restore checkpoint image %s: %v", resolvedImageName, r.Err)
```

Comment:
This should either be 125 or 126.
```
man podman run
...
Exit Status
       The exit code from podman run gives information about why the container failed to run or why it exited. When podman run exits with a non-zero code, the exit codes follow the chroot(1) standard, see below:

       125 The error is with Podman itself

              $ podman run --foo busybox; echo $?
              Error: unknown flag: --foo
              125

       126 The contained command cannot be invoked

              $ podman run busybox /etc; echo $?
              Error: container_linux.go:346: starting container process caused "exec: \"/etc\": permission denied": OCI runtime error
              126

       127 The contained command cannot be found

              $ podman run busybox foo; echo $?
              Error: container_linux.go:346: starting container process caused "exec: \"foo\": executable file not found in $PATH": OCI runtime error
              127

       Exit code contained command exit code

              $ podman run busybox /bin/sh -c 'exit 3'; echo $?
              3
```

Reply:
I've updated the exit code to 126. Thanks!